### PR TITLE
Feature/GitHub pages

### LINKED ISSUE #354: Read the Docs build failing
Recent builds of the docs are failing: https://readthedocs.org/projects/pymalcolm/builds/

This should be used as an opportunity to migrate to GitHub pages.

### PR DESCRIPTION
**Pull request by BenBradnick**
Migrating to GitHub pages... this fixes #354

### CODE REVIEW

**Comment by BenBradnick** [comment]
Seems to be working - has a deployment here: http://dls-controls.github.io/pymalcolm/merge/index.html (I assume master will be built after merging, I copied the docs CI from scanspec)

File: docs/conf.py [hunk 1]
```diff
@@ -104,29 +91,30 @@ def setup(app):
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 # on_rtd is whether we are on readthedocs.org
-on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
+on_rtd = os.environ.get("READTHEDOCS", None) == "True"
```
Comment (thomascobb) [inline docs/conf.py:94]: You can get rid of this now

File: docs/conf.py [hunk 2]
```diff
@@ -104,29 +91,30 @@ def setup(app):
...
 
 if not on_rtd:  # only import and set the theme if we're building docs locally
     import sphinx_rtd_theme
+
     html_context = dict(css_files=[])
-    html_theme = 'sphinx_rtd_theme'
+    html_theme = "sphinx_rtd_theme"
```
Comment (thomascobb) [inline docs/conf.py:100]: I recommend https://gitlab.diamond.ac.uk/controls/templates/dls_python3_template_module/-/blob/master/%7B%7Bcookiecutter.project_name%7D%7D/docs/conf.py#L100

File: docs/conf.py [hunk 3]
```diff
@@ -159,28 +145,30 @@ def setup(app):
 # (source start file, target name, title,
 #  author, documentclass [howto, manual, or own class]).
 latex_documents = [
-    ('contents', 'malcolm.tex', u'malcolm Documentation',
-     u'Tom Cobb', 'manual'),
+    ("contents", "malcolm.tex", u"malcolm Documentation", u"Tom Cobb", "manual"),
```
Comment (thomascobb) [inline docs/conf.py:148]: This can all go, in fact you could replace most of this file with conf.py from the template module

**Pull request merged**